client/doh.js

```diff
@@ -29,23 +29,23 @@ const DOHClient = ({ dns } = {}) => {
     if (recursive) {
       packet.header.rd = 1;
     }
-    if (clientIp) {
-      query.additionals.push(Packet.Resource.EDNS([
-        Packet.Resource.EDNS.ECS(clientIp)
-      ]));
-    };
     packet.questions.push({
       name,
-      class: cls,
-      type: Packet.TYPE[type],
+      class : cls,
+      type  : Packet.TYPE[type],
     });
     const query = packet.toBase64URL();
+    if (clientIp) {
+      query.additionals.push(Packet.Resource.EDNS([
+        Packet.Resource.EDNS.ECS(clientIp),
+      ]));
+    }
```

Comment:
Remove this block, because “query” is String

Reply:
Updated both mentions, can you check the block?